test/Bridges/Constraint/indicator_sos.jl

```diff
@@ -0,0 +1,168 @@
+    @test MOI.get(model, MOI.ConstraintSet(), bridge3) == iset3
+    @test MOI.get(model, MOI.ConstraintFunction(), bridge3) ≈ f3
+end
+
+@testset "Basic constraint test" begin
+    mock = MOIU.MockOptimizer(MOIU.UniversalFallback(MOIU.Model{Float64}()))
+    config = MOIT.TestConfig()
+    for BC in [MOI.LessThan{Float64}, MOI.GreaterThan{Float64}]
+        bridged_mock = MOIB.Constraint.IndicatortoSOS1{Float64, BC, MOI.ConstraintIndex{MOI.SingleVariable, BC}}(mock)
+        MOIT.basic_constraint_tests(bridged_mock, config,
+            include=[
+                (MOI.VectorAffineFunction{Float64}, MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE, BC}),
+            ],
+        )
+    end
+end
+
+@testset "Model equality" begin
+    mock = MOIU.MockOptimizer(MOIU.UniversalFallback(MOIU.Model{Float64}()))
+    config = MOIT.TestConfig()
+    bridged_mock = MOIB.Constraint.IndicatortoSOS1{Float64, MOI.LessThan{Float64}, MOI.ConstraintIndex{MOI.SingleVariable, MOI.LessThan{Float64}}}(mock)
+    (z, bin_cons) = MOI.add_constrained_variable(bridged_mock, MOI.ZeroOne())
+    MOI.set(bridged_mock, MOI.VariableName(), z, "z")
+    MOI.set(mock, MOI.ConstraintName(), bin_cons, "bin_cons")
+    x = MOI.add_variable(bridged_mock)
+    MOI.set(bridged_mock, MOI.VariableName(), x, "x")
+    var_names = ["z", "x", "w"]
+    f = MOI.VectorAffineFunction(
+        [MOI.VectorAffineTerm(1, MOI.ScalarAffineTerm(1.0, z)),
+         MOI.VectorAffineTerm(2, MOI.ScalarAffineTerm(1.0, x)),
+        ],
+        [0.0, 0.0]
+    )
+    iset = MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE}(MOI.LessThan(8.0))
+    ci = MOI.add_constraint(bridged_mock, f, iset)
+    @test length(MOI.get(mock, MOI.ListOfVariableIndices())) == 3
+    MOI.set(mock, MOI.VariableName(), MOI.get(mock, MOI.ListOfVariableIndices()), var_names)
+    sos_cons_list = MOI.get(mock, MOI.ListOfConstraintIndices{MOI.VectorOfVariables, MOI.SOS1{Float64}}())
+    @test length(sos_cons_list) == 1
+    MOI.set(mock, MOI.ConstraintName(), sos_cons_list[1], "sos1")
+    single_less_cons = MOI.get(mock, MOI.ListOfConstraintIndices{MOI.SingleVariable, MOI.LessThan{Float64}}())
+    @test length(single_less_cons) == 1
+    MOI.set(mock, MOI.ConstraintName(), single_less_cons[1], "wless")
+
+    inequality_list = MOI.get(mock, MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}}())
+    @test length(inequality_list) == 1
+    MOI.set(mock, MOI.ConstraintName(), inequality_list[1], "ineq")
+    MOI.set(mock, MOI.ObjectiveFunction{MOI.SingleVariable}(), MOI.SingleVariable(z))
+    MOI.set(mock, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    s = """
+    variables: z, x, w
+    sos1: [w, z] in MathOptInterface.SOS1([0.4, 0.6])
+    wless: w <= 0.0
+    ineq: x + w <= 8.0
+    bin_cons: z in MathOptInterface.ZeroOne()
+    maxobjective: z
+    """
+    model = MOIU.Model{Float64}()
+    MOIU.loadfromstring!(model, s)
+    MOIU.test_models_equal(mock, model, var_names, ["sos1", "wless", "ineq", "bin_cons"])
+
```

Comment:
We should also use `test_model_equals` on `bridged_mock`. It also tests that ListOfConstraintIndices, ... are implemented correctly as well as ConstraintFunction, ...

Reply:
you mean testing against:
```julia
s = """
variables: z, x
indicator_cons: [z, 1.0 * x] in MathOptInterface.IndicatorSet{MathOptInterface.ACTIVATE_ON_ONE}(MathOptInterface.LessThan{Float64}(8.0))
bin_cons: z in MathOptInterface.ZeroOne()
maxobjective: z
"""

```

Reply:
done